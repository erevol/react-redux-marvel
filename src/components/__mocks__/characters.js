export const characters = {
    isLoading: false,
    items: [
        {
        id: 1011334,
        name: '3-D Man',
        description: '',
        modified: '2014-04-29T14: 18: 17-0400',
        thumbnail: {
          path: 'http: //i.annihil.us/u/prod/marvel/i/mg/c/e0/535fecbbb9784',
          extension: 'jpg'
            },
        resourceURI: 'http: //gateway.marvel.com/v1/public/characters/1011334',
        comics: {
          available: 2,
          collectionURI: 'http: //gateway.marvel.com/v1/public/characters/1011334/comics',
          items: [
                    {
              resourceURI: 'http: //gateway.marvel.com/v1/public/comics/21366',
              name: 'Avengers: The Initiative (2007) #14'
                    },
                    {
              resourceURI: 'http: //gateway.marvel.com/v1/public/comics/24571',
              name: 'Avengers: The Initiative (2007) #14 (SPOTLIGHT VARIANT)'
                    }
                ],
          returned: 2
            },
        series: {
          available: 2,
          collectionURI: 'http: //gateway.marvel.com/v1/public/characters/1011334/series',
          items: [
                    {
              resourceURI: 'http: //gateway.marvel.com/v1/public/series/1945',
              name: 'Avengers: The Initiative (2007 - 2010)'
                    },
                    {
              resourceURI: 'http: //gateway.marvel.com/v1/public/series/2005',
              name: 'Deadpool (1997 - 2002)'
                    }
                ],
          returned: 2
            },
        stories: {
          available: 2,
          collectionURI: 'http: //gateway.marvel.com/v1/public/characters/1011334/stories',
          items: [
                    {
              resourceURI: 'http: //gateway.marvel.com/v1/public/stories/19947',
              name: 'Cover #19947',
              type: 'cover'
                    },
                    {
              resourceURI: 'http: //gateway.marvel.com/v1/public/stories/19948',
              name: 'The 3-D Man!',
              type: 'interiorStory'
                    }
                ],
          returned: 2
            },
        events: {
          available: 1,
          collectionURI: 'http: //gateway.marvel.com/v1/public/characters/1011334/events',
          items: [
                    {
              resourceURI: 'http: //gateway.marvel.com/v1/public/events/269',
              name: 'Secret Invasion'
                    }
                ],
          returned: 1
            },
        urls: [
                {
            type: 'detail',
            url: 'http: //marvel.com/comics/characters/1011334/3-d_man?utm_campaign=apiRef&utm_source=7fa1221c73cec1a6cd06c58d6400f61e'
                },
                {
            type: 'wiki',
            url: 'http: //marvel.com/universe/3-D_Man_(Chandler)?utm_campaign=apiRef&utm_source=7fa1221c73cec1a6cd06c58d6400f61e'
                },
                {
            type: 'comiclink',
            url: 'http: //marvel.com/comics/characters/1011334/3-d_man?utm_campaign=apiRef&utm_source=7fa1221c73cec1a6cd06c58d6400f61e'
                }
            ]
        },
        {
        id: 1017100,
        name: 'A-Bomb (HAS)',
        description: 'Rick Jones has been Hulk\'s best bud since day one, but now he\'s more than a friend...he\'s a teammate! Transformed by a Gamma energy explosion, A-Bomb\'s thick, armored skin is just as strong and powerful as it is blue. And when he curls into action, he uses it like a giant bowling ball of destruction! ',
        modified: '2013-09-18T15: 54: 04-0400',
        thumbnail: {
          path: 'http: //i.annihil.us/u/prod/marvel/i/mg/3/20/5232158de5b16',
          extension: 'jpg'
            },
        resourceURI: 'http: //gateway.marvel.com/v1/public/characters/1017100',
        comics: {
          available: 1,
          collectionURI: 'http: //gateway.marvel.com/v1/public/characters/1017100/comics',
          items: [
                    {
              resourceURI: 'http: //gateway.marvel.com/v1/public/comics/40632',
              name: 'Hulk (2008) #53'
                    }
                ],
          returned: 1
            },
        series: {
          available: 1,
          collectionURI: 'http: //gateway.marvel.com/v1/public/characters/1017100/series',
          items: [
                    {
              resourceURI: 'http: //gateway.marvel.com/v1/public/series/3374',
              name: 'Hulk (2008 - 2012)'
                    }
                ],
          returned: 1
            },
        stories: {
          available: 1,
          collectionURI: 'http: //gateway.marvel.com/v1/public/characters/1017100/stories',
          items: [
                    {
              resourceURI: 'http: //gateway.marvel.com/v1/public/stories/92078',
              name: 'Hulk (2008) #55',
              type: 'cover'
                    }
                ],
          returned: 1
            },
        events: {
          available: 0,
          collectionURI: 'http: //gateway.marvel.com/v1/public/characters/1017100/events',
          items: [],
          returned: 0
            },
        urls: [
                {
            type: 'detail',
            url: 'http: //marvel.com/characters/76/a-bomb?utm_campaign=apiRef&utm_source=7fa1221c73cec1a6cd06c58d6400f61e'
                },
                {
            type: 'comiclink',
            url: 'http: //marvel.com/comics/characters/1017100/a-bomb_has?utm_campaign=apiRef&utm_source=7fa1221c73cec1a6cd06c58d6400f61e'
                }
            ]
        }
    ]
}