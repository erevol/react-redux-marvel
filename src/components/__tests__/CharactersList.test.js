import CharacterList from '../CharactersList';
import {charactersMock} from '../__mocks__/characters';
import {characterMock} from '../__mocks__/character';

it("shallows correctly", () => {
    const wrapper = shallow(
        <CharacterList characters={charactersMock} character={characterMock} fetchAllCharacters={() => {}} fetchCharacterById={() => {}} />
    );

    expect(wrapper).toMatchSnapshot();
});