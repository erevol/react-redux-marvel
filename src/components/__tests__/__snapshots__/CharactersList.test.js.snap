// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shallows correctly 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
